### DISCUSSION TITLE
MQTT to Home Assistant

### DISCUSSION THREAD

**Discussion by Flatericuk** [teslamate-org/teslamate#2942] [Getting started]
Apologies in advance, I'm very new and this is probably very easy but I'm struggling. I have Teslamate installed through Docker on a Pi and I'm trying to get the data in HA which is installed on a virtual machine. I just can't work out how to get MQTT working in this instance.

I'm sure the answer is in the Teslamate docs but I can't for the life of me work out how to point TM to the MQTT instance in HA

Any assistance greatly appreciated

**Comment by Kriss1670** [comment]
I'm a newby too but here is the point where I am.
You first need to install Mosquitto broker Add-on on HA (not the MQTT integration) and configure it with a username / password :
```
- username: myuser
  password: mypassword
```
Then save and start the add-on.

Then you need to edit the docker-compose.yml file on your Pi : `nano docker-compose.yml` and add the keys  MQTT_HOST / MQTT_USER  / MQTT_PASSWORD with the IP adress of Home Assistant and the credentials used in HA. Then restart Docker `docker-compose up -d` or restart your Pi.
Then you should have a valid connection which you can check by installing in HA the add-on MQTT Explorer and connect to your HA Broker (HA IP Address + credentials)
